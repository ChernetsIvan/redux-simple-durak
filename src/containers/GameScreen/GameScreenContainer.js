import { connect } from "react-redux";

import GameScreen from "./../../components/GameScreen/GameScreen";

import { clickOnBeginGameButton } from "./../../actions";

const mapStateToProps = store => ({
  aiField: store.game.aiField,
  playerField: store.game.playerField
});

export const mapDispatchToProps = dispatch => ({
  onBeginGameClick: () => {
    dispatch(clickOnBeginGameButton());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(GameScreen);
