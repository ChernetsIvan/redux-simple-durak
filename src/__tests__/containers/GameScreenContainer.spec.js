import React from "react";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { createStore } from "redux";

import GameScreenContainer, {
  mapDispatchToProps
} from "./../../containers/GameScreen/GameScreenContainer";
import reducer from "./../../reducers";

describe("GameScreenContainer snapshot testing", () => {
  let store;

  beforeEach(() => {
    store = createStore(reducer);
  });

  test("Store in initial-state mode", () => {
    const component = renderer.create(
      <Provider store={store}>
        <GameScreenContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("Map Dispatch To Props", () => {
  test("should call onBeginGameClick action", () => {
    const mockFunction = jest.fn(() => "Mock function");
    const { onBeginGameClick } = mapDispatchToProps(mockFunction);
    onBeginGameClick();
    expect(mockFunction).toBeCalled();
  });
});
