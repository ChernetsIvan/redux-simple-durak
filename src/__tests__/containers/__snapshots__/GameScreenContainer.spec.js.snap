// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreenContainer snapshot testing Store in initial-state mode 1`] = `
<div>
  <div
    className="container  my-1"
  >
    <div
      className="row justify-content-between"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          Новая игра
        </button>
      </div>
      <div
        className="col-auto"
      >
        <h3
          className="mt-2"
        >
          <span>
            —
          </span>
           
          Ваш ход!
        </h3>
      </div>
      <div
        className="col-2"
      />
    </div>
  </div>
  <div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-1"
        />
        <div
          className="col-10"
        >
          <div
            className="container invisible"
          >
            <div
              className="row"
            >
              <div
                className="col-1 card mb-1 shirt"
              />
            </div>
          </div>
        </div>
        <div
          className="col-1"
        />
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-10"
        >
          <div
            className="container invisible"
          >
            <div
              className="row"
            >
              <div
                className="col-1 card mb-1 shirt"
              />
            </div>
          </div>
          <div
            className="container invisible"
          >
            <div
              className="row"
            >
              <div
                className="col-1 card mb-1 shirt"
              />
            </div>
          </div>
        </div>
        <div
          className="col-2"
        >
          <div
            className="container"
          >
            <div
              className="row"
            />
            <div
              className="row"
            >
              <div
                className="col-6"
              />
              <div
                className="col-6"
              >
                <div
                  className="container"
                >
                  <div
                    className="row justify-content-center"
                  >
                    <div
                      className="col-auto"
                    >
                      <b />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-1"
        />
        <div
          className="col-10"
        >
          <div
            className="container invisible"
          >
            <div
              className="row"
            >
              <div
                className="col-1 card mb-1 shirt"
              />
            </div>
          </div>
        </div>
        <div
          className="col-1"
        />
      </div>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-auto"
        />
      </div>
    </div>
  </div>
</div>
`;
